pkg/plugins/manager/signature/manifest.go

```diff
@@ -339,15 +339,46 @@ func (s *Signature) Verify(ctx context.Context, keyID string, block *clearsign.B
 		return fmt.Errorf("%v: %w", "failed to parse public key", err)
 	}
 
+	signatureBody, err := io.ReadAll(block.ArmoredSignature.Body)
+	if err != nil {
+		return fmt.Errorf("%v: %w", "failed to read signature body", err)
+	}
+
 	if _, err = openpgp.CheckDetachedSignature(keyring,
 		bytes.NewBuffer(block.Bytes),
-		block.ArmoredSignature.Body, &packet.Config{}); err != nil {
-		// If the key includes revocations, we can assume that the key was revoked
-		if len(keyring) > 0 && len(keyring[0].Revocations) > 0 {
-			return fmt.Errorf("%s (KeyID: %s): %w", openpgpErrors.ErrKeyRevoked.Error(), keyID, err)
+		bytes.NewBuffer(signatureBody), &packet.Config{}); err != nil {
+		if errors.Is(err, openpgpErrors.ErrKeyRevoked) {
+			valid, err := wasSignedBeforeRevocation(keyring, signatureBody)
```

Comment:
Nit. Easier to read perhaps.

```suggestion
			signedBeforeRevocation, err := wasSignedBeforeRevocation(keyring, signatureBody)
```